base/file.jl

```diff
@@ -824,23 +824,23 @@ julia> readdir(abspath("base"), join=true)
 """
 function readdir(dir::AbstractString; join::Bool=false, sort::Bool=true)
     # Allocate space for uv_fs_t struct
-    uv_readdir_req = zeros(UInt8, ccall(:jl_sizeof_uv_fs_t, Int32, ()))
+    req = Libc.malloc(_sizeof_uv_fs)
```

Comment:
This would require extra error handling (currently missing) to do try/finally/Libc.free—and this is typically slower. The rest of the PR seems good.

Reply:
Good catch. It looks like `truncate` and `futime` are also missing this error handling.  Should I split these into a separate PR or add them here?

Reply:
those are theoretically written not to fail, so they didn't need additional error handling

Reply:
I timed these changes with master with BenchmarkTools, and for both directories with a small number of files and directories with a large number of files  the timing are within < 2% for small directories (which makes sense) (this PR reduces a single allocation). This suggests the `try,catch` handling is not the main cost here, which makes sense.